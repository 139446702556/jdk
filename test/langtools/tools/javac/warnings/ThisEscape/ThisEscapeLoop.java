/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @compile/ref=ThisEscapeLoop.out -Xlint:this-escape -XDrawDiagnostics ThisEscapeLoop.java
 * @summary Verify 'this' escape detection properly handles loop convergence
 */

public class ThisEscapeLoop {

    public ThisEscapeLoop() {
        ThisEscapeLoop ref1 = this;
        ThisEscapeLoop ref2 = null;
        ThisEscapeLoop ref3 = null;
        ThisEscapeLoop ref4 = null;
        for (int i = 0; i < 100; i++) {
            ref4 = ref3;
            ref3 = ref2;
            ref2 = ref1;
            if (ref4 != null)
                ref4.mightLeak();
        }
    }

    public void mightLeak() {
    }
}
