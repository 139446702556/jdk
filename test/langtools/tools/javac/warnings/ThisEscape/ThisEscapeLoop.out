ThisEscapeLoop.java:20:31: compiler.warn.possible.this.escape
1 warning
